dashboard/app/controllers/datablock_storage_controller.rb

```diff
@@ -0,0 +1,323 @@
+# DatablockStorage is the backend for the 'Data' tab of Applab, the 'Data' blocks
+# and the dataset browser. DatablockStorage stores student-controlled data generated
+# by and with the data blocks like `createRecord`, `updateRecord`, `setKeyValue`, etc.
+#
+# These backend methods are accessed by a thin wrapper on the JS side: datablockStorage.js
+#
+# Student data is stored in corresponding ActiveRecord models:
+# - DatablockStorageTable: stores a list of tables and their columns
+# - DatablockStorageRecord: stores the records in each table
+# - DatablockStorageKvp: stores key-value pairs
+#
+# Metadata for the code.org defined datasets are stored in:
+# - DatablockStorageLibraryManifest
+
+class DatablockStorageController < ApplicationController
+  before_action :validate_channel_id
+  before_action :authenticate_user!
+  skip_before_action :verify_authenticity_token
+
+  StudentFacingError = DatablockStorageTable::StudentFacingError
+
+  rescue_from StudentFacingError do |exception|
+    render json: {msg: exception.message, type: exception.type}, status: :bad_request
+  end
+
+  SUPPORTED_PROJECT_TYPES = ['applab', 'gamelab']
```

Comment:
I thought gamelab wasn't supported yet?

Reply:
hrm, yeah, I can see how this is confusing. It's supported, but it's not enabled yet. 😅

Reply:
The missing piece for gamelab is just that we haven't turned it on in gamelab itself. Mainly because we'd prefer to start with as small a test as possible.

Reply:
I guess maybe my question is why gamelab is added here but not turned on. But I also think that's a nit and not a blocker :)

Reply:
Yeah, it's going to get turned on soon